ui/pages/settings/contact-list-tab/edit-contact/edit-contact.component.js

```diff
@@ -74,27 +75,38 @@ export default class EditContact extends PureComponent {
           className="settings-page__header address-book__header--edit"
           paddingLeft={6}
           paddingRight={6}
+          width={BlockSize.Full}
+          alignItems={AlignItems.center}
         >
-          <Box display={Display.Flex} alignItems={AlignItems.center}>
+          <Box
+            display={Display.Flex}
+            alignItems={AlignItems.center}
+            style={{ overflow: 'hidden' }}
+            paddingRight={2}
+          >
             <AvatarAccount size={AvatarAccountSize.Lg} address={address} />
             <Text
               className="address-book__header__name"
               variant={TextVariant.bodyLgMedium}
               marginInlineStart={4}
+              style={{ overflow: 'hidden' }}
+              ellipsis
             >
               {name || address}
             </Text>
           </Box>
-          <Button
-            type="link"
-            className="settings-page__address-book-button"
-            onClick={async () => {
-              await removeFromAddressBook(chainId, address);
-              history.push(listRoute);
-            }}
-          >
-            {t('deleteContact')}
-          </Button>
+          <Box className="settings-page__address-book-button">
+            <Button
```

Comment:
non-blocking: it would help if we can replace deprecated code with component-library in files that are updated

![Screenshot 2024-06-11 at 8 27 36 AM](https://github.com/MetaMask/metamask-extension/assets/8112138/40825222-37c7-4c72-99fb-a824d9fe2d64)

Reply:
I'll replace once we get the `BadgeIcon` in, thanks!